scripts/components/api-changes-validator/api_report_parser.ts

```diff
@@ -0,0 +1,23 @@
+import ts from 'typescript';
+
```

Comment:
no test for `ApiReportParser`?

Reply:
It's covered by `scripts/components/api-changes-validator/api_usage_generator.test.ts` below.

It also runs as part of `scripts/components/api-changes-validator/api_changes_validator.ts` integration test and also runs as part of GH Action check.

While working on this PR I was trying to find balance between providing adequate coverage and local debugability vs writing tests for the sake of having them. This particular component is a this wrapper on top of 3rd party TS API, if it doesn't work 3 other places will go red.

In general this PR relies heavy on typescript compiler (both api and command line) hence testing effort is integration heavy (but still fast, all scripts tests complete <5s ).